### PR TITLE
add `-check` flag to fmt

### PR DESCRIPTION
**Pull request by MattSurabian**
This PR is meant to handle #3817 and add check support to the fmt command. There's not much to it but in local testing with waypoint.hcl files it seemed to perform as expected, though it's kinda funny that the default config is generated needing formatting.

```
matt$ ./waypoint fmt -h
Usage: waypoint fmt [options] [FILE]

  Rewrite a waypoint.hcl file to a canonical format.

  This only works for HCL-formatted Waypoint configuration files. JSON-formatted
  files do not work and will result in an error.

  If FILE is not specified, then the current directory will be searched
  for a "waypoint.hcl" file. If FILE is "-" then the content will be read
  from stdin.

  This command does not validate the waypoint.hcl configuration. This will
  work for older and newer configuration formats.

Global Options:

  -app=<string>
      App to target. Certain commands require a single app target for Waypoint
      configurations with multiple apps. If you have a single app, then this
      can be ignored. This is aliased as "-a".

  -plain
      Plain output: no colors, no animation. The default is false.

  -project=<string>
      Project to target. This is aliased as "-p".

  -workspace=<string>
      Workspace to operate in. This is aliased as "-w".

Command Options:

  -check
      Check if the input is formatted. Exit status will be 0 if all input is
      properly formatted and non-zero otherwise. The default is false.

  -write
      Overwrite the input file. If this is false, the formatted output will be
      written to STDOUT. This has no effect when formatting from STDIN or when
      using the -check flag. The default is true.

nick-cage:waypoint on 3817-fmt-check 
matt$ ./waypoint init
Do you want help generating a waypoint.hcl file? Type 'yes' to initialize the interactive generator or 'no' to generate a template waypoint.hcl file: no

Generating template file

Initial Waypoint configuration created!
No Waypoint configuration was found in this directory.

A sample configuration has been created in the file "waypoint.hcl". This
file is heavily commented to help you get started.

Once you've setup your initial configuration, run "waypoint init" again to
validate the configuration and initialize your project.


nick-cage:waypoint on 3817-fmt-check 
matt$ ./waypoint fmt -check waypoint.hcl 


nick-cage:waypoint on 3817-fmt-check 
matt$ echo $?
3

nick-cage:waypoint on 3817-fmt-check 
matt$ ./waypoint fmt
waypoint.hcl

nick-cage:waypoint on 3817-fmt-check 
matt$ ./waypoint fmt -check waypoint.hcl 

nick-cage:waypoint on 3817-fmt-check 
matt$ echo $?
0
```

### CODE REVIEW

File: internal/cli/fmt.go [hunk 1]
```diff
@@ -108,7 +122,15 @@ func (c *FmtCommand) Flags() *flag.Sets {
...
-				"from STDIN.",
+				"from STDIN or when using the -check flag.",
+		})
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "check",
+			Target:  &c.flagCheck,
+			Default: false,
+			Usage: "Check if the input is formatted. Exit status will be 0 if " +
+				"all input is properly formatted and non-zero otherwise.",
```
Comment (briancain) [inline internal/cli/fmt.go:133]:
```suggestion
				"all input is properly formatted and exit status 3 otherwise.",
```

**Review by briancain — CHANGES_REQUESTED** [review]
Thanks for adding this flag to our `fmt` command! I've got a small doc string suggestion for you, but otherwise this makes sense!

**Comment by MattSurabian** [comment]
Suggestion makes sense to me, I almost went for that initially but opted for the language in the issue instead. I think this is definitely clearer. :+1:

**Review by briancain — APPROVED** [review]
Looks great to me! Tested it locally and it works.

I went ahead and made a changelog and auto-generated the website for you, however since this PR was opened on your fork, I don't think I have push permissions. If you wanted to cherry-pick my commits then I'd say we're good to merge this one! It's over on this branch: https://github.com/hashicorp/waypoint/compare/4020/check-flag-fmt?expand=1

**Review by briancain — APPROVED** [review]
Looks good to me! Thank you for making this contribution! 🎉

**Pull request merged**